### DISCUSSION TITLE
OpenTelemetry instrumentation

### DISCUSSION THREAD

**Discussion by tanenbaum** [ClickHouse/clickhouse-go#1267] [General]
Hello!

I was looking at how I might get instrumentation out of the library, both traces and metrics. I noticed that the ch-go "low level" API instruments various code paths, like https://github.com/ClickHouse/ch-go/blob/main/query.go#L609. I did at first think this library used those code paths, but as far as I can tell it is only sharing the `driver.Conn` interface definition and not using the ch-go client.

Is there any plan to do the same sort of instrumentation in this library? Would you like contributions in this area?

**Comment by jkaflik** [comment]
@tanenbaum absolutely! We don't have it in our roadmap, but any contribution is welcomed.

**Reply by tanenbaum** [reply]
Excellent. We may start with a wrapper around the driver interface in the short term and then look to see how it could be properly integrated here.
